### PR TITLE
chore:splitstore: Update SplitStore Readme

### PR DESCRIPTION
**Pull request by TippyFlitsUK**
Updated SplitStore Readme to show changes in https://github.com/filecoin-project/lotus/pull/9571

## Related Issues

https://github.com/filecoin-project/lotus/pull/9571

## Proposed Changes

Updates to readme file

## Checklist

Before you mark the PR ready for review, please make sure that:

- [X] Commits have a clear commit message.
- [X] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [X] New features have usage guidelines and / or documentation updates in
  - [X] [Lotus Documentation](https://lotus.filecoin.io)
  - [X] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [X] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: blockstore/splitstore/README.md [hunk 1]
```diff
@@ -3,129 +3,111 @@
...
-for small nodes that are simply watching the chain.
-
-*Warning:* Using the discard store for a general purpose node is discouraged, unless
-you really know what you are doing. Use it at your own risk.
-
 ## Configuration Options
 
-These are options in the `[Chainstore.Splitstore]` section of the configuration:
-
-- `HotStoreType` -- specifies the type of hotstore to use.
```
Comment (ZenGround0) [inline blockstore/splitstore/README.md:42]: I think only the ColdStoreType config value is out of date.  All the rest are more thorough than the config comments and still relevant so if they are still relevant I'd support keeping them around.

File: blockstore/splitstore/README.md [hunk 2]
```diff
@@ -3,129 +3,111 @@
...
-- We collect cold objects by iterating through the hotstore and checking the mark set; if an object is not marked, then it is candidate for purge.
+
+- We prepare a transaction whereby all i/o referenced objects through the API are tracked.
+- We walk the chain and mark reachable objects, keeping four finalities of state roots and messages and all headers all the way to genesis.
+- Once the chain walk is complete, we begin full transaction protection with concurrent marking; we walk and mark all references created during the chain walk. At the same time, all I/O through the API concurrently marks objects as live references.
+- We collect cold objects by iterating through the hotstore and checking the mark set; if an object is not marked, then it is a candidate for purge.
 - When running with a coldstore, we next copy all cold objects to the coldstore.
-- At this point we are ready to begin purging:
-  - We sort cold objects heaviest first, so as to never delete the consituents of a DAG before the DAG itself (which would leave dangling references)
```
Comment (ZenGround0) [inline blockstore/splitstore/README.md:105]: Nice catch removing sorting.
The "We delete in small batches ..." line is still relevant and should be kept around

**Review by ZenGround0 — APPROVED** [review]
LGTM. Thank for this.

File: blockstore/splitstore/README.md [hunk 3]
```diff
@@ -3,129 +3,111 @@
...
+  # env var: LOTUS_CHAINSTORE_SPLITSTORE_HOTSTOREMESSAGERETENTION
+  HotStoreMessageRetention = 0
+
+  # HotStoreFullGCFrequency specifies how often to perform a full (moving) GC on the hotstore.
+  # A value of 0 disables, while a value 1 will do full GC in every compaction.
+  # Default is 20 (about once a week).
+  #
+  # type: uint64
+  # env var: LOTUS_CHAINSTORE_SPLITSTORE_HOTSTOREFULLGCFREQUENCY
+  HotStoreFullGCFrequency = 20
```
Comment (jennijuju) [inline blockstore/splitstore/README.md:68]: does this mean 20 finalities?

File: blockstore/splitstore/README.md [hunk 4]
```diff
@@ -3,129 +3,111 @@
...
-In general you _should not_ have to use the discard store, unless you
-are running a network assistive node (like a bootstrapper or booster)
-or have very constrained hardware with not enough disk space to
-maintain a coldstore, even with garbage collection. It is also appropriate
-for small nodes that are simply watching the chain.
-
-*Warning:* Using the discard store for a general purpose node is discouraged, unless
-you really know what you are doing. Use it at your own risk.
-
 ## Configuration Options
```
Comment (jennijuju) [inline blockstore/splitstore/README.md:30]: can we explain how user should set each of these configuration base on their needs in detail? (a link to the lotus.docs.io is fine here

**Review by jennijuju — CHANGES_REQUESTED** [review]

File: blockstore/splitstore/README.md [hunk 5]
```diff
@@ -3,129 +3,111 @@
...
-and reliably archive out of scope objects in a coldstore.  The
-coldstore can also be a discard store, whereby out of scope objects
-are discarded or a regular badger blockstore (the default), which can
-be periodically garbage collected according to configurable user
-retention policies.
+With lotus v1.19.0, we introduce the next iteration in design and
+implementation, which we call SplitStore v2.
+
+The new design (see [#9056](https://github.com/filecoin-project/lotus/pull/9056) and [#9128](https://github.com/filecoin-project/lotus/discussions/9128))
+evolves the splitstore to be a freestanding compacting blockstore that allows you to keep a small 60 GiB to 275 GiB working set in a hot blockstore and reliably archive out-of-scope objects in a coldstore. The coldstore can also be a `discard` store, whereby out-of-scope objects are discarded, a `universal` store, which will store all chain data or a `messages` store which will only store on-chain messages. The `messaqges` badger blockstore is the default storage type.
```
[the remark this answers is not in the document]
Comment (jennijuju) [reply blockstore/splitstore/README.md:10]: > The `messaqges` badger blockstore is the default storage type.

is `messaqges` a typo here?

File: blockstore/splitstore/README.md [hunk 6]
```diff
@@ -3,129 +3,111 @@
...
-allows us to keep a small (60-100GB) working set in a hot blockstore
-and reliably archive out of scope objects in a coldstore.  The
-coldstore can also be a discard store, whereby out of scope objects
-are discarded or a regular badger blockstore (the default), which can
-be periodically garbage collected according to configurable user
-retention policies.
+With lotus v1.19.0, we introduce the next iteration in design and
+implementation, which we call SplitStore v2.
+
+The new design (see [#9056](https://github.com/filecoin-project/lotus/pull/9056) and [#9128](https://github.com/filecoin-project/lotus/discussions/9128))
```
[the remark this answers is not in the document]
Comment (jennijuju) [reply blockstore/splitstore/README.md:9]:
```
// CompactionThreshold is the number of epochs that need to have elapsed
	// from the previously compacted epoch to trigger a new compaction.
	//
	//        |················· CompactionThreshold ··················|
	//        |                                             |
	// =======‖≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡‖------------------------»
	//        |                    |  chain -->             ↑__ current epoch
	//        | archived epochs ___↑
	//                             ↑________ CompactionBoundary
	//
	// === :: cold (already archived)
	// ≡≡≡ :: to be archived in this compaction
	// --- :: hot
```

this graph might be useful

[the remark this answers is not in the document]
Comment (TippyFlitsUK) [reply blockstore/splitstore/README.md:9]: Thanks Jen - Done

**Pull request closed without merging**